### ISSUE TITLE
Allow users to register a callback method that will be invoked on every new event(s) received

### ISSUE THREAD

**Issue by mzesri** [Azure/azure-event-hubs#22]
Implementing the ability of client receiver(s) to register to receive new events/messages with the Java API using Java futures seems unnatural for use and makes the client SDK user application need to implement a very cumbersome code (and also potentially needing to create multiple receiving threads, one for each partition, in order to simplify the implementation).  It seems much more natural to allow users to register a callback method that will be invoked on every new event(s) received.  This is a more reactive and easy to use pattern to use and implement by client users, for continuously receiving a stream of events/messages reactively upon event, compared to a future/promise approach that is more sutural for a one time operation type of case, or a set of tasks that need to executed and finish.
In other words, I would strongly recommend modifying the client implementation to allow users to simply register a callback method for receiving messages.

**Comment by JamesBirdsall** [comment]
You are absolutely correct, that is a much more natural pattern for most use cases. We've had support for that pattern in the .NET world for a while now with the EventProcessorHost client, and we are now in the process of porting EventProcessorHost to Java. It will sit on top of the low-level client and provide almost exactly what you describe.

**Comment by SreeramGarlapati** [comment]
PR#25: https://github.com/Azure/azure-event-hubs/pull/25 includes this feature in the eventhubs JavaClient SDK...
